### PR TITLE
Added new details for getstakinginfo error message as suggested in #288

### PR DESCRIPTION
**Pull request by francisjyap**
Added new details for getstakinginfo error message as suggested in #288

Errors messages are :

* Warning: Wallet is locked. Please enter the wallet passphrase with walletpassphrase first.
* Warning: We don't appear to have mature coins.

### CODE REVIEW

**Comment by aguycalled** [comment]
utACK

File: src/main.cpp [hunk 1]
```diff
@@ -7120,7 +7142,7 @@ bool static ProcessMessage(CNode* pfrom, string strCommand, CDataStream& vRecv,
         CTransaction tx;
         vRecv >> tx;
 
-	LogPrint("net", "Received tx %s peer=%d\n%s\n", tx.GetHash().ToString(), pfrom->id, tx.ToString());
+    LogPrint("net", "Received tx %s peer=%d\n%s\n", tx.GetHash().ToString(), pfrom->id, tx.ToString());
```
Comment (aguycalled) [inline src/main.cpp]: wrong indent

File: src/main.h [hunk 2]
```diff
@@ -289,6 +289,7 @@ bool IsInitialBlockDownload();
  * This function only returns the highest priority warning of the set selected by strFor.
  */
 std::string GetWarnings(const std::string& strFor);
+std::string GetWarnings(const std::string& strFor, bool verbose);
```
Comment (aguycalled) [inline src/main.h]: I'd consider changing the variable name to something more descriptive like fForStaking. The warnings returned when the flag is false are not less verbose than the staking-related.

**Comment by aguycalled** [comment]
Built and tested on OSX. Added some comments about minor suggestions

**Comment by mxaddict** [comment]
Built and tested using `Ubuntu 18.04`, I agree with the comments from @aguycalled 

@francisjyap Go Go Go! :100:

**Comment by francisjyap** [comment]
@aguycalled Changed the variable name as per your suggestion. Also fixed the wrong indent.

**Comment by mxaddict** [comment]
Built and tested on Ubuntu 18.04 will pass the review once travis build passes.

**Comment by aguycalled** [comment]
travis failed because of timeout

**Pull request merged**